indykite/data_source_application.go

```diff
@@ -129,36 +179,63 @@ func dataApplicationListContext(ctx context.Context, data *schema.ResourceData,
 	}
 	ctx, cancel := context.WithTimeout(ctx, data.Timeout(schema.TimeoutRead))
 	defer cancel()
-	resp, err := clientCtx.GetClient().ListApplications(ctx, &configpb.ListApplicationsRequest{
-		AppSpaceId: data.Get(appSpaceIDKey).(string),
-		Match:      match,
-	})
+
+	// Get project ID from either project_id or app_space_id (backward compatibility)
+	projID := data.Get(projectIDKey).(string)
+	if projID == "" {
+		projID = data.Get(appSpaceIDKey).(string)
+	}
+
+	// Validate that at least one of app_space_id or project_id is provided
+	if projID == "" {
+		return diag.Diagnostics{
+			{
+				Severity: diag.Error,
+				Summary:  "Missing required argument: one of `app_space_id` or `project_id` must be specified",
+			},
+		}
+	}
```

Comment:
Hint: this can be done on Terraform validation level, and not during `terraform apply`

Reply:
updated